addon/initializers/component-styles.js

```diff
@@ -0,0 +1,36 @@
+import Ember from 'ember';
+import podNames from 'ember-component-css/pod-names';
+
+const {
+  Component,
+  ComponentLookup,
+} = Ember;
+
+export function initialize() {
+
+  ComponentLookup.reopen({
+    componentFor(name, owner) {
+      if (podNames[name] && !owner.application.hasRegistration('component:' + name)) {
```

Comment:
Also, wouldn't it be easier to call super and check if the return value is undefined instead of manually checking registration like this? Something like:

    let FoundComponent = this._super(...arguments);
    If (podNames[name] && !FoundComponent) { 
      FoundComponent = Component;
      owner.register('component:' + name, FoundComponent);
    }
    
    return FoundComponent;